lib/puppet/reports/servicenow.rb

```diff
@@ -16,7 +16,17 @@ def process
 
     Puppet.info(sn_log_entry("incident creation conditions: #{incident_creation_conditions}"))
 
-    unless create_incident?(status, resource_statuses, incident_creation_conditions)
+    # Some incident creation conditions depend on the resource events. Thus, we go ahead
+    # and evaluate these 'event' conditions _before_ evaluating the incident creation
+    # conditions so we can iterate through the resource events only once. This results in
+    # cleaner code for a negligible performance hit, where the performance hit occurs if the
+    # incident creation conditions do _not_ contain an 'event' condition.
+    #
+    # Note that event_conditions is a hash of <condition> => <satisfied?>
+    event_conditions = event_conditions(resource_statuses)
```

Comment:
Hmm, why not inline this in the `satisfied_conditions` method? Feels like that makes more sense since event conditions are part of calculating satisfied conditions.

Reply:
I can move it in there if we feel like that's cleaner, but I was thinking the event conditions would likely be handy to have available for the incident down the road.

Reply:
I don't see any particular reason to move this. It works fine where it is.

Reply:
The reason is b/c `event_conditions` here makes it seem like it's a separate thing from `satisfied_conditions` when it isn't -- `event_conditions` was just an intermediate variable that was added so we iterate through the resource events only once. So, intermediate calculation variable instead of a sub-abstraction.

But moving this is also not a PR blocker so if y'all would prefer it as-is then I'm OK with that.

Reply:
Note main change I'm saying is just do

```
calculate_satisfied_conditions(status, resource_statuses, incident_creation_conditions)
```

and have this function's implementation call `event_conditions` method.

Reply:
That sounds reasonable, updated!